crawler/processing/cptd_processor.py

```diff
@@ -0,0 +1,152 @@
+
+        - If any feedback for an expected UUID has errors, raises a CherrypickerDataError with a message to try again.
+        - If any of the UUIDs don't receive feedback within a reasonable time frame, a CherrypickerDataError will be
+          raised with a message to try again.
+
+        Arguments:
+           message_uuids {list} -- a list of message UUIDs to look for feedback on.
+        """
+        LOGGER.info(
+            "Beginning loop of feedback parsing for cherrypicker test data. Loop will run continuously until no "
+            f"message is seen for {self._config.CPTD_FEEDBACK_WAIT_TIME} seconds."
+        )
+        unconfirmed_uuids = message_uuids.copy()
+        t_end = time() + self._config.CPTD_FEEDBACK_WAIT_TIME
+
+        with BasicGetter(
+            general_helpers.get_rabbit_server_details(
+                self._config, self._config.RABBITMQ_CPTD_USERNAME, self._config.RABBITMQ_CPTD_PASSWORD
+            )
+        ) as basic_getter:
+            while time() < t_end and len(unconfirmed_uuids) > 0:
+                fetched_message = basic_getter.get_message(self._config.RABBITMQ_CPTD_FEEDBACK_QUEUE)
+                if fetched_message is None:
+                    sleep(1)
+                    continue  # There was no message on the queue
+
+                t_end = time() + self._config.CPTD_FEEDBACK_WAIT_TIME  # Extend our loop time
+
+                rabbit_message = RabbitMessage(fetched_message.headers, fetched_message.body)
+                if rabbit_message.subject != RABBITMQ_SUBJECT_CREATE_PLATE_FEEDBACK:
+                    LOGGER.debug(
+                        f"Fetched message has incorrect subject '{rabbit_message.subject}'. Discarding message."
+                    )
+                    continue  # We can only process create plate feedback messages
+
+                rabbit_message.decode(self._decoder)
+                if not rabbit_message.contains_single_message:
+                    LOGGER.debug("Fetched message contains more than one feedback message. Discarding message.")
+                    continue  # We can only process single messages
+
+                feedback_message = CreatePlateFeedbackMessage(rabbit_message.message)
+                source_uuid = feedback_message.source_message_uuid.value
+                if source_uuid not in unconfirmed_uuids:
+                    LOGGER.debug("Fetched message is for an unrecognised UUID. Discarding message.")
+                    continue  # We aren't interested in any messages for other UUIDs
+
+                error_free = feedback_message.operation_was_error_free.value
+                if not error_free:
+                    LOGGER.info(
+                        f"Cherrypicker test data create message with UUID '{source_uuid}' was not processed without "
+                        "errors."
+                    )
+                    break
+
+                unconfirmed_uuids.remove(source_uuid)
+
```

Comment:
I would refactor this section of code. continue could create problems in future. The output condition is modified dynamically and there are too many breaking points (continue, break).

Reply:
Could you elaborate on the problems you'd expect continue to cause?  I'm happy to refactor by extracting logic into private functions if you think it's needed, but I can't see how this could be done without use of continue and break nor whether it would actually make it simpler to follow.

Reply:
breaks and continues change the flow of the program. I'm not saying not to use it; I'm only saying that this function has a lot of them and could potentially originate future errors when modifying the code (indentation, inner loops, ...). Thinking about other ways of doing it without continues and breaks, it seems to me a flow where any failing step send you to the beginning of the process again, one possibility could be a pipeline pattern? ```process([step1, step2, step3,...])```

Reply:
I've modified how we approach the parsing of the feedback message reducing us down to a single continue for a completely missing message and a break when a message we were expecting contains errors.  Is this more appealing or do you want to work on another solution?